File: app/src/main/java/com/android/app/fancytextreplacer/utils/CopyHandler.java
package com.android.app.fancytextreplacer.utils;

import android.content.ClipData;
import android.content.ClipboardManager;
import android.content.Context;
import android.content.Intent;
import android.widget.Toast;

public class CopyHandler {
    Context context;

    public CopyHandler(Context context) {
        this.context = context;
    }

    public void copyText(String data) {
        if (data.isEmpty()) {
            Toast.makeText(context, "Enter some text", Toast.LENGTH_SHORT).show();
            return;
        }
        ClipboardManager clipboardManager = (ClipboardManager) context.getSystemService(Context.CLIPBOARD_SERVICE);
        Toast.makeText(context, "Copied to clipboard! Your copied text is\n" + data, Toast.LENGTH_SHORT).show();
        ClipData clip = ClipData.newPlainText("simple text", data);
        if (clipboardManager != null) {
            clipboardManager.setPrimaryClip(clip);
        }
    }

    public void shareText(String data) {
        if (data.isEmpty()) {
            Toast.makeText(context, "Enter some text", Toast.LENGTH_SHORT).show();
            return;
        }
        try {
            Intent i = new Intent(Intent.ACTION_SEND);
            i.setType("text/plain");
            i.putExtra(Intent.EXTRA_TEXT, data);
            context.startActivity(Intent.createChooser(i, "choose one"));
        } catch (Exception e) {
            //e.toString();
        }
    }
}

File: app/src/main/java/com/stylish/fancy/text/generator/utils/CopyHandler.kt
package com.stylish.fancy.text.generator.utils

import android.content.ClipData
import android.content.ClipboardManager
import android.content.Context
import android.content.Intent
import android.widget.Toast

class CopyHandler(var context: Context) {

    fun copyText(data: String) {
        if (data.isEmpty()) {
            Toast.makeText(context, "Enter some text", Toast.LENGTH_SHORT).show()
            return
        }
        val clipboardManager =
            context.getSystemService(Context.CLIPBOARD_SERVICE) as ClipboardManager
        Toast.makeText(context, "Copied to clipboard!", Toast.LENGTH_SHORT).show()
        val clip = ClipData.newPlainText("simple text", data)
        clipboardManager.setPrimaryClip(clip)
    }

    fun shareText(data: String) {
        if (data.isEmpty()) {
            Toast.makeText(context, "Enter some text", Toast.LENGTH_SHORT).show()
            return
        }
        try {
            val i = Intent(Intent.ACTION_SEND)
            i.type = "text/plain"
            i.putExtra(Intent.EXTRA_TEXT, data)
            context.startActivity(Intent.createChooser(i, "choose one"))
        } catch (e: Exception) {
            e.printStackTrace()
        }
    }
}